crates/subspace-networking/src/behavior/custom_record_store/providers.rs

```diff
@@ -30,6 +31,15 @@ pub trait EnumerableProviderStorage<'a> {
     fn known_providers(&'a self) -> Self::ProvidedIter;
 }
 
+/// Defines operation for a storage provider with fixed records.
+pub trait FixedProviderRecordStorage: Clone {
+    /// Returns true if the storage contains a records with given key.
+    fn has_fixed_provider_record(&self, key: &Key) -> bool;
+
+    /// Register key as local fixed provider record.
+    fn register_fixed_local_provider(&self, key: &Key);
+}
```

Comment:
I do not understand what this is and why it is necessary. Neither commit message not commit in the code explain this.